rust/agama-lib/src/product/client.rs

```diff
@@ -146,6 +146,64 @@ impl<'a> ProductClient<'a> {
         Ok(addons)
     }
 
+    // details of available addons
+    pub async fn available_addons(&self) -> Result<Vec<AddonProperties>, ServiceError> {
+        let addons: Vec<AddonProperties> = self
+            .registration_proxy
+            .available_addons()
+            .await?
+            .into_iter()
+            .map(|hash| AddonProperties {
+                id: hash
+                    .get("id")
+                    .unwrap_or(&zbus::zvariant::Value::new(""))
+                    .try_into()
+                    .unwrap_or_default(),
```

Comment:
get optional property is not possible to use? https://github.com/agama-project/agama/blob/master/rust/agama-lib/src/dbus.rs#L51 ... also all those unwrap_or_default looks bad. It means it can hide errors in empty string.

Reply:
Basically if property is mandatory it should not use empty string if it is not there....and if it is not mandatory then type should be `Option<String>`...regarding the last `unwrap_or_default` it should be simple `?` to propagate error.

Reply:
example usage is at https://github.com/agama-project/agama/blob/master/rust/agama-server/src/storage/web/iscsi.rs#L106